src/sparse-merkle-tree/sparse-merkle-tree.js

```diff
@@ -47,9 +47,16 @@ function setNodeValue(db, key, value, prefix) {
 * @returns {Buffer} - Key node value
 */
 function getHashFinalNode(hi, hv) {
-  const hashArray = [hi, hv];
+  // const hashArray = [hi, hv];
   // const hashKey = mimc7.multiHash(hashArray, bigInt(1));
-  const hashKey = poseidon.multiHash(hashArray, bigInt(1));
+  // const hashKey = poseidon.multiHash(hashArray, bigInt(1));
+
+  // let toH = [];
+  // toH.push(bigInt(1));
+  // toH.push(hashArray);
+  const hashArray = [bigInt(1), hi, hv];
```

Comment:
extra element for hash final node should be at the right of the array:
`const hashArray = [hi, hv, bigInt.one];`

Reply:
I'm checking this and seems that in the internal claim's spec & in the go version we have `[hi, hv, 1]`, while in the circomlib there is `[1, hi, hv]`
Updated `iden3js` & `go-iden3-core` to match with the `circomlib` version. Added `sparse-merkle-tree` tests to check same output than in `circomlib` in `iden3jssrc/canary.test.js`

Reply:
When the data is stored in `circomlib` library is stored as `[1, key, value]` but when it is hashed it uses `hash1` function which hashes array `[key, value, bigInt.one]`

Reply:
also, updated the **claims spec**, from `[1, hi, hv]` to `[hi, hv, 1]`
After DC5 we should:
- put the `claims spec` into the doc website (docs.iden3.io)
- add test vector generation for go&js/ts&rust test code to match it with circomlib version

Reply:
Great !!